src/main/java/io/dropwizard/revolver/core/resilience/ResilienceUtil.java

```diff
@@ -195,47 +197,39 @@ private static void updateBulkheadsForApiConfigs(RevolverServiceConfig revolverS
                                     revolverServiceConfig.getService() + "." + revolverHttpApiConfig.getApi();
                             log.info("ThreadPool Name : {}, Concurrency : {} ", threadPoolName,
                                     hystrixCommandConfig.getThreadPool().getConcurrency());
-                            updateBulkheadRegistry(hystrixCommandConfig.getThreadPool(),
-                                    threadPoolName);
+                            poolVsBulkHead.putIfAbsent(threadPoolName,
+                                    getBulkheadWithConcurrency(hystrixCommandConfig.getThreadPool().getConcurrency(),
+                                            revolverServiceConfig.getService()));
                         }
                     });
         }
     }
 
-    private static void updateBulkheadRegistry(ThreadPoolConfig threadPoolConfig, String threadPoolName) {
-        if (POOL_VS_BULK_HEAD.get(threadPoolName) == null) {
-            POOL_VS_BULK_HEAD.put(threadPoolName, bulkheadRegistry.bulkhead(
-                    threadPoolName,
-                    BulkheadConfig.custom().maxConcurrentCalls(
-                            threadPoolConfig
-                                    .getConcurrency())
-                            .build()));
-        } else {
-            POOL_VS_BULK_HEAD.put(threadPoolName,
-                    bulkheadRegistry.replace(
-                            threadPoolName,
-                            Bulkhead.of(threadPoolName,
-                                    BulkheadConfig.custom().maxConcurrentCalls(
-                                            threadPoolConfig
-                                                    .getConcurrency())
-                                            .build())).get());
-        }
-    }
-
     private static void updateBulkheadsForThreadPools(
+            Map<String, Bulkhead> poolVsBulkHead,
             RevolverServiceConfig revolverServiceConfig) {
         ThreadPoolGroupConfig threadPoolGroupConfig = revolverServiceConfig.getThreadPoolGroupConfig();
         if (threadPoolGroupConfig != null) {
             threadPoolGroupConfig.getThreadPools().forEach(threadPoolConfig -> {
                 String threadPoolName =
                         getThreadPoolName(revolverServiceConfig, threadPoolConfig);
                 log.info("ThreadPool Name : {}, Concurrency : {} ", threadPoolName, threadPoolConfig.getConcurrency());
-                updateBulkheadRegistry(threadPoolConfig, threadPoolName);
+                poolVsBulkHead.putIfAbsent(threadPoolName,
+                        getBulkheadWithConcurrency(threadPoolConfig.getConcurrency(), threadPoolName));
             });
 
         }
     }
 
+    private static Bulkhead getBulkheadWithConcurrency(int concurrency, String threadPoolName) {
+        bulkheadRegistry.remove(threadPoolName);
```

Comment:
This is not threadSafe. If at the same time, another request is trying to get bulkHead from registry for same threadpool, it will get null